src/rules.c

```diff
@@ -326,7 +322,7 @@ SchemaPrefixNode *SchemaPrefixNode_Create(const char *prefix, IndexSpec *index)
 }
 
 void SchemaPrefixNode_Free(SchemaPrefixNode *node) {
```

Comment:
Unrelated
The function itself isn't used at the moment.
should we make both `SchemaPrefixNode_Create` and `SchemaPrefixNode_Free` static?

Reply:
Yes, looks like the SchemaPrefixNode "type" should be used only inside rules.c.
However, note the `freePrefixNode()` callback that also frees a `SchemaPrefixNode`, only that it does that right.
I would suggest moving the logic of `freePrefixNode()` into `SchemaPrefixNode_Free()`, making it inline, and have `freePrefixNode()` call `SchemaPrefixNode_Free()`.
This looks like passing coins from left pocket to the right, but it will make more sense when c++ comes (and `freePrefixNode()` becomes a lambda).